build/Toolset/Toolset.csproj

```diff
@@ -12,10 +12,8 @@
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'">
-  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
```

Comment:
Adding/Editing/Removing files doesn't cause regeneration, nor does changing things in the `Library` property page (such as AssemblyName or RootNamespace).

Changing something in the `Build` property page (which are all of the centrally defined properties we have in `Settings.props` or in the global properties page) will cause this to expand, but expands it with the modified property contained there-in (this is the case even if you modify both Config|Platform groups simultaneously).